src/test/java/io/confluent/idesidecar/restapi/util/WebClientFactoryTest.java

```diff
@@ -1,14 +1,98 @@
 package io.confluent.idesidecar.restapi.util;
 
 
+import static com.github.tomakehurst.wiremock.client.WireMock.aResponse;
+import static com.github.tomakehurst.wiremock.client.WireMock.urlEqualTo;
+import static com.github.tomakehurst.wiremock.core.WireMockConfiguration.options;
+import static org.junit.jupiter.api.Assertions.assertEquals;
+
+import com.github.tomakehurst.wiremock.WireMockServer;
+import com.github.tomakehurst.wiremock.client.WireMock;
+import com.github.tomakehurst.wiremock.matching.UrlPattern;
+import io.confluent.idesidecar.restapi.application.SidecarInfo;
+import io.quarkiverse.wiremock.devservice.ConnectWireMock;
 import io.quarkus.test.junit.QuarkusTest;
+import io.quarkus.test.junit.mockito.InjectSpy;
+import io.vertx.core.Vertx;
+import io.vertx.ext.web.client.WebClient;
+import io.vertx.ext.web.client.WebClientOptions;
 import jakarta.inject.Inject;
+import org.junit.jupiter.api.AfterEach;
 import org.junit.jupiter.api.Assertions;
+import org.junit.jupiter.api.BeforeEach;
 import org.junit.jupiter.api.Test;
 
 @QuarkusTest
+@ConnectWireMock
 public class WebClientFactoryTest {
   @Inject WebClientFactory webClientFactory;
+  WireMock wireMock;
+  WireMockServer wireMockServer;
+
+  @InjectSpy
+  SidecarInfo sidecarInfo;
+
+  @BeforeEach
+  public void setup() {
+    wireMockServer = new WireMockServer(options().dynamicPort());
+    wireMockServer.start();
+    WireMock.configureFor("localhost", wireMockServer.port());
+  }
+  @AfterEach
+  void resetWireMock() {
+    wireMockServer.stop();
+    wireMock.removeMappings();
+  }
+
+  @Test
+  void getDefaultWebClientOptionsIncludesUserAgent() {
+    var userAgent = webClientFactory.getDefaultWebClientOptions().getUserAgent();
+    Assertions.assertTrue(userAgent.contains("support@confluent.io) sidecar/"));
```

Comment:
Quarkus provides us a nice way of injecting specific application configuration overrides for test classes with the use of the [`@QuarkusTestProfile`](https://quarkus.io/guides/getting-started-testing#writing-a-profile) annotation. (Take a look at the [`NoAccessFilterProfile`](https://github.com/confluentinc/ide-sidecar/blob/main/src/test/java/io/confluent/idesidecar/restapi/testutil/NoAccessFilterProfile.java) and its usages for example). 

For this test, I believe we could hardcode some realistic values for the configuration that `SidecarInfo` expects (`vscode.version`, `os.version` and so on) by defining a new test profile class ([see StackOverflow](https://stackoverflow.com/a/69267852)). With that, we should be able to assert the expected value of _full_ user-agent string.

Reply:
@rohitsanj , IIUC test profiles work well for configurations that are already specified at build or runtime, like `vscode.version`, but it's not possible to specify values for system properties like `os.name`, etc. If my understanding is correct, then the test would still have to mock the `SidecarInfo` methods like `osType()` and `osArch()` anyway, so it'd be maybe simpler to also mock `vsCode()` rather than define a whole new test profile or even use an existing test profile and rely upon a combination of mock and test-profile configured values. Thoughts?

Reply:
Oh, that's really good to know. Thanks for trying it out!